gpkit/keydict.py

```diff
@@ -121,9 +123,8 @@ def __setitem__(self, key, value):
         key, idx = self.parse_and_index(key)
         if key not in self.keymap:
             self.keymap[key].add(key)
-            if hasattr(key, "keys") and self.keymapping:
-                for mapkey in key.keys:
-                    self.keymap[mapkey].add(key)
+            if hasattr(key, "keys"):
+                self._unmapped_keys.add(key)
```

Comment:
interesting, I expected the laziness to occur on `__getitem__`, such that I would only map keys for entries I get (e.g. if I call `__getitem__("W")`, then I add `"W"` to `_unmapped_keys`. Doesn't this cause the entire map to get built during SolutionArray construction (which uses `__setitem__`) when that isn't necessary?